clcache.py

```diff
@@ -1612,19 +1618,28 @@ def processCompileRequest(cache, compiler, args):
     printOutAndErr(out, err)
     return exitCode
 
+def formBaseCommandLine(cmdLine, sourceFiles):
+    # type: (List[str], List[Tuple[str, str]]) -> List[str]
+    setOfSources = set([sourceFile for sourceFile, _ in sourceFiles])
+    skippedArgs = ('/MP', '/Tc', '/Tp')
+    baseCmdLine = [
+        arg for arg in cmdLine
+        if not (arg in setOfSources or arg.startswith(skippedArgs))
+    ]
+
+    return baseCmdLine
+
 def scheduleJobs(cache, compiler, cmdLine, environment, sourceFiles, objectFiles):
-    baseCmdLine = []
-    setOfSources = set(sourceFiles)
-    for arg in cmdLine:
-        if not (arg in setOfSources or arg.startswith("/MP")):
-            baseCmdLine.append(arg)
+    # type: (???, str, List[str], ???, List[Tuple[str, str]], List[str]) -> int
+    # Filter out all source files from the command line to form baseCmdLine
+    baseCmdLine = formBaseCommandLine(cmdLine, sourceFiles)
 
     exitCode = 0
     cleanupRequired = False
     with concurrent.futures.ThreadPoolExecutor(max_workers=jobCount(cmdLine)) as executor:
         jobs = []
-        for srcFile, objFile in zip(sourceFiles, objectFiles):
-            jobCmdLine = baseCmdLine + [srcFile]
+        for srcFile, srcLanguage, objFile in zip(zip(*sourceFiles), objectFiles):
+            jobCmdLine = baseCmdLine + list(filter(None, [srcLanguage, srcFile]))
```

Comment:
Since `srcLanguage` is either an empty string or one of `/Tp` or `/Tc`, couldn't you drop the composition of `list` and `filter` and just use

```python
jobCmdLine = baseCmdLine + [srcLanguage + srcFile]
```

Reply:
Will do.